allennlp/tests/training/no_op_trainer_test.py

```diff
@@ -0,0 +1,31 @@
+import os
+from typing import Dict
+
+import torch
+
+from allennlp.common.testing import AllenNlpTestCase
+from allennlp.data import Vocabulary
+from allennlp.data.dataset_readers import SequenceTaggingDatasetReader
+from allennlp.models.model import Model
+from allennlp.training import NoOpTrainer
+
+
+class ConstantModel(Model):
+    def forward(self, *inputs) -> Dict[str, torch.Tensor]:
+        return {"class": torch.LongTensor(1)}
+
+class TestNoOpTrainer(AllenNlpTestCase):
+    def setUp(self):
+        super().setUp()
+        self.instances = SequenceTaggingDatasetReader().read(self.FIXTURES_ROOT / 'data' / 'sequence_tagging.tsv')
+        vocab = Vocabulary.from_instances(self.instances)
+        self.vocab = vocab
+        self.model = ConstantModel(vocab)
+
+    def test_trainer_serializes(self):
+        serialization_dir = self.TEST_DIR / "serialization_dir"
+        trainer = NoOpTrainer(serialization_dir=serialization_dir, model=self.model)
+        metrics = trainer.train()
+        assert metrics == {}
+        assert os.path.exists(serialization_dir / "best.th")
+        assert os.path.exists(serialization_dir / "vocabulary")
```

Comment:
similarly, do you want to test here that the archived model is "correct"?

Reply:
This I would prefer to elide. The correctness is guaranteed by the end-to-end test in `no_op_train_test.py` and the tests in `checkpointer_test.py`. Anything more here would be duplication.